docs/csharp/programming-guide/classes-and-structs/local-functions.md

```diff
@@ -0,0 +1,76 @@
+---
+title: "Local functions (C# Programming Guide) | Microsoft Docs"
+ms.date: "2017-06-14"
+ms.prod: .net
+ms.technology: 
+  - "devlang-csharp"
+ms.topic: "article"
+dev_langs: 
+  - "CSharp"
+helpviewer_keywords: 
+  - "fields [C#]"
+author: "rpetrusha"
+ms.author: "ronpet"
+---
+# Local functions (C# Programming Guide)
+
+Starting with C# 7, C# supports *local functions*. Local functions are private methods of a type that are nested in another member. They can only be called from their containing member. Local functions can be declared in and called from:
+
+- Methods, especially iterator methods and async methods
+- Constructors
+- Property accessors
+- Event accessors
+- Anonymous methods
+- Lambda expressions
+- Finalizers
+- Other local functions
+
+However, local functions can't be declared inside an expression-bodied member.
+
+> [!NOTE]
+> In some cases, you can use a lambda expression to implement functionality also supported by a lambda expression. For a comparison, see [Local functions compared to Lambda expressions](../../local-functions-vs-lambdas.md).
+
+Local functions make the intent of your code clear. Anyone reading you code can see that the method is private and is not callable except by the containing method. For team projects, they also make it impossible for another developer to mistakenly call the method directly from elsewhere in the class or stuct.
+ 
+## Local function syntax
+
+A local function is defined as a nested method inside a containing member. Its definition has the following syntax:
+
+```csharp
+<return-value> <method-name> <parameter-list>
+```
+
+Unlike a method definition, a local function definition cannot include the following elements:
+
+- The [private](..\..\language-reference\keywords\private.md) member access modifier. Because all local functions are private, including the `private` keyword generates compiler error CS0106, "The modifier 'private' is not valid for this item."
+ 
+- The [static](..\..\language-reference\keywords\static.md) keyword. Whether a local function is static or instance depends on its enclosing member. Including the `static` keyword generates compiler error CS0106, "The modifier 'static' is not valid for this item."
```

Comment:
As I already mentioned on the CS0106 article, local functions also can't use many other modifiers. I *think* the only valid modifiers on local functions are `async` and `unsafe`, so maybe it would be easier to list those here?

Also, a local function (including its parameters and type parameters) can't have attributes, that might be worth mentioning here.